ExceptionsHomework/src/ro/scit/menu/Menu.java

```diff
@@ -0,0 +1,160 @@
+                        LocalDate currentDate = LocalDate.now();
+                        System.out.println("The age is : " );
+                        int age = DateOfBirth.calculateAge(birthDate, currentDate );
+                        System.out.println(age);
+                        System.out.println("Insert the gender: male or female");
+                        String gender = sc.next();
+                        if (gender.equalsIgnoreCase("male")) {
+                            student.setGender(GenderEnum.MALE.getGender());
+                        } else if (gender.equalsIgnoreCase
+                                ("female")) {
+                            student.setGender(GenderEnum.FEMALE.getGender());
+                        } else {
+                            System.out.println("Invalid option! Please try again.");
+                        }
+                        Student student1 = new Student(cnp,firstName, lastName, birthDate, age, gender);
+                        addStudent(student1);
+                        System.out.println("Your Student has been added");
+                    }catch (NumberFormatException e){
+                        System.err.println("Please insert a number");
+                    }
+
+//                    try {
+//                        Date date = sdf.parse(str);
+//                        sdf = new SimpleDateFormat("dd.MM.yyyy");
+//                        System.out.println("Date of birth: " + sdf.format(date));
+//                    } catch (ParseException e) {
+//                        System.out.println("Parse Exception");
+//                    }
+
+
+                    break;
+                case 2:
+                    System.out.println("Insert the student's cnp");
+                    String cnp1 = sc.nextLine();
+                    deleteStudent(cnp1);
+                    System.out.println("The student has been deleted! ");
+
+                    break;
+                case 3:
+                    System.out.println("Insert the age: ");
+                    int age = Integer.parseInt(sc.nextLine());
+                    System.out.println("all students with age " + age + " : ");
+                    retrievebyAge(age);
+                    break;
+                case 4:
+                    System.out.println("List Sorted by name");
+                    StudentNameComparator nameCompare = new StudentNameComparator();
+                    Collections.sort(studentList, nameCompare);
+                    for (Student s: studentList)
+                        System.out.println(s);
+                    break;
+                case 5:
+                    this.exit();
+                    break;
+                default:
+                    System.out.println("Invalid selection.");
+                    break;
+
+            }
+        }
+    }
+
+    private void exit () {
+        System.out.println("Exiting...");
+        System.exit(1);
```

Comment:
It's not a problem to exit with code 1, but usually this means error and I think in this case it's considered normal flow. I'd exit with code 0 instead.